tests/test_visualization/test_histogram.py

```diff
@@ -20,80 +20,73 @@ def setUp(self):
             'annotation2': np.random.choice(annotation_types, size=100),
         }, index=cell_range)
         var = pd.DataFrame(index=['marker1', 'marker2', 'marker3'])
-        self.adata = anndata.AnnData(X, obs=annotation, var=var)
+        self.adata = anndata.AnnData(
+            X.astype(np.float32), obs=annotation, var=var
+        )
+
+    def test_both_feature_and_annotation(self):
+        err_msg = ("Cannot pass both feature_name and "
+                   "annotation_name, choose one")
+        with self.assertRaisesRegex(ValueError, err_msg):
+            histogram(
+                self.adata,
+                feature_name='marker1',
+                annotation_name='annotation1'
+            )
```

Comment:
quote: "[@fangliu117](https://github.com/fangliu117) , would you please add that annotation_name can't be combined with feature_name?"
@georgezakinih I moved this validation test at the beginning of test code. Do you think this test is good enough to address your comment? or I need to expand more? Thank you~